model/layer.js

```diff
@@ -29,7 +68,60 @@ class Layer extends ParchKey {
      * @returns {boolean} true if both name and id are valid, false otherwise.
      */
     validate() {
-        return super.validate();
+        let valid = super.validate();
+
+        // Components
+        if (this.components.length === 0) {
+            valid = false;
+            console.log('Layer (' + this.id + '): Field "components" cannot be empty.');
+        }
+
+        this.components.forEach((value, index) => {
+            if (!value.validate()) {
+                valid = false;
+                console.log('Layer (' + this.id + '): Field "components" contains an invalid Component object:\n\tID: '
+                        + value.id + '\n\tIndex: ' + index);
+            }
+        });
+
+
+        // Connections
+        if (this.connections.length === 0) {
+            valid = false;
+            console.log('Layer (' + this.id + '): Field "connections" cannot be empty.');
+        }
```

Comment:
So you can technically have no component and/or no connections on a layer, so I wouldn't classify this as a failure, but it is something I would emit a warning about. Perhaps modify this check and the component length check into a single check like this and put it before checking the components or connections.

```js
if (this.components.length === 0 && this.connections.length == 0) {
    console.log('Layer (' + this.id + '): contains no component and no connections');
}
else if (this.components.length == 0) {
    console.log('Layer (' + this.id + '): contains no component');
}
else if (this.connections.length == 0) {
    console.log('Layer (' + this.id + '): contains no connections');
}
```

Reply:
Good to know thank you. It's been changed.